week10/src/pages/Detail.jsx

```diff
@@ -0,0 +1,392 @@
+    const [commentsList, setCommentsList] = useState(null);
+    const [myAccount, setMyAccount] = useState("");
+    const nav = useNavigate();
+
+    useEffect(() => {
+        async function getDetails() {
+            const res = await GetPostDetailApi(postId);
+            setDetails(res);
+            const res2 = await JWTtest();
+
+            setMyAccount(res2.nickname);
+        }
+
+        async function getHeartNumInDetail() {
+            const num = await getHeartNum(postId);
+            setHeartNum(num);
+        }
+
+        async function getCommentsList() {
+            const res = await getComments(postId);
+            setCommentsList(res);
+        }
+
+        async function didIClickHeart() {
+            const HeartHistory = await checkMyClick(postId);
+            setMyheart(HeartHistory);
+        }
+
+        if (!localStorage.getItem("efubtoken")) {
+            alert("로그인 후 이용해주세요.");
+            nav("/register");
+        } else {
+            getDetails();
+            getHeartNumInDetail();
+            getCommentsList();
+            didIClickHeart();
+        }
+    }, [postId]);
+
+    useEffect(() => {
+        async function updateHeartNum() {
+            const num = await getHeartNum(postId);
+            setHeartNum(num);
+        }
+
+        updateHeartNum();
+    }, [myHeart, postId]);
+
+    if (!details) {
+        return <div>로딩 중...</div>;
+    }
+
+    const handleEditBtn = () => {
+        setIsEdit(true);
+    };
+
+    const onSubmitEditPost = async () => {
+        if (post.title.trim() === "") {
+            alert("제목을 입력해주세요.");
+        } else if (post.content.trim() == "") {
+            alert("내용을 입력해주세요.");
+        } else {
+            await EditPostApi(postId, post);
+            setIsEdit(false);
+            window.location.reload();
+        }
+    };
+
+    const onChangesPost = (e) => {
+        const { name, value } = e.target;
+        setPost({ ...post, [name]: value });
+    };
+
+    const deletePost = async () => {
+        await DeletePostApi(postId);
+        alert("게시글 삭제 완료");
+        nav(-1);
+    };
+
+    const onChangesMyComments = (e) => {
+        const { value } = e.target;
+        setMyComments(value);
+    };
+
+    const onSubmitMyComments = async (e) => {
+        if (myComments.trim() === "") {
+            alert("내용을 입력해주세요.");
+        } else {
+            const data = {
+                postId: postId,
+                content: myComments,
+            };
+            await postComments(data);
+            window.location.reload();
+        }
+    };
+
+    const onClickDeleteComments = async (commentId) => {
+        console.log("실행 ", postId);
+        await deleteComments(commentId);
+        window.location.reload();
+    };
+
+    const onClickHeart = async () => {
+        if (myHeart) {
+            await deleteHeart(postId);
+        } else {
+            await postHeart(postId);
+        }
+        setMyheart(!myHeart);
```

Comment:
```suggestion
        setMyheart(!myHeart);
        const num = await getHeartNum(postId);
        setHeartNum(num);
```
사용자가 좋아요 상태를 변경했다면 좋아요 개수를 다시 가져오는 코드는 이 부분에 추가해주면 됩니다!